### PR TITLE
App config : phased rollout support for terraform

### PR DESCRIPTION
**Pull request by vhavalagi**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

image

image

Output from acceptance testing:

image

### CODE REVIEW

File: ibm/service/appconfiguration/data_source_ibm_app_config_features.go [hunk 1]
```diff
@@ -119,6 +119,11 @@ func DataSourceIBMAppConfigFeatures() *schema.Resource {
 							Computed:    true,
 							Description: "Tags associated with the feature.",
 						},
+						"rollout_percentage": {
+							Type:        schema.TypeInt,
+							Optional:    true,
```
Comment (hkantare) [inline ibm/service/appconfiguration/data_source_ibm_app_config_features.go:124]: it shd be computed

File: ibm/service/appconfiguration/data_source_ibm_app_config_features.go [hunk 2]
```diff
@@ -152,6 +157,11 @@ func DataSourceIBMAppConfigFeatures() *schema.Resource {
 										Computed:    true,
 										Description: "Order of the rule, used during evaluation. The evaluation is performed in the order defined and the value associated with the first matching rule is used for evaluation.",
 									},
+									"rollout_percentage": {
+										Type:        schema.TypeInt,
+										Optional:    true,
```
Comment (hkantare) [inline ibm/service/appconfiguration/data_source_ibm_app_config_features.go:162]: same as above

File: ibm/service/appconfiguration/resource_ibm_app_config_feature.go [hunk 3]
```diff
@@ -161,7 +171,7 @@ func resourceIbmIbmAppConfigFeatureCreate(d *schema.ResourceData, meta interface
 	options.SetEnabledValue(d.Get("enabled_value").(string))
 	options.SetEnvironmentID(d.Get("environment_id").(string))
 	options.SetDisabledValue(d.Get("disabled_value").(string))
-
+	options.SetRolloutPercentage(int64(d.Get("rollout_percentage").(int)))
```
Comment (hkantare) [inline ibm/service/appconfiguration/resource_ibm_app_config_feature.go:174]: rollout_percentage is optional so you need to call d.GetOk(...) and then set the value if its defined

Comment (hkantare) [reply ibm/service/appconfiguration/resource_ibm_app_config_feature.go:174]:
```
if _, ok := d.GetOk("rollout_percentage"); ok {
			options.SetRolloutPercentage(int64(d.Get("rollout_percentage").(int)))
		}
		```

File: ibm/service/appconfiguration/data_source_ibm_app_config_feature.go [hunk 4]
```diff
@@ -78,6 +78,11 @@ func DataSourceIBMAppConfigFeature() *schema.Resource {
 				Computed:    true,
 				Description: "Denotes if the targeting rules are specified for the feature flag.",
 			},
+			"rollout_percentage": {
+				Type:        schema.TypeInt,
```
[the remark this answers is not in the document]
Comment (vhavalagi) [reply ibm/service/appconfiguration/data_source_ibm_app_config_feature.go:82]: Corrected it to computed

File: ibm/service/appconfiguration/data_source_ibm_app_config_features.go (hunk 1 above)
Comment (vhavalagi) [reply ibm/service/appconfiguration/data_source_ibm_app_config_features.go]: Corrected it

File: ibm/service/appconfiguration/resource_ibm_app_config_feature.go (hunk 3 above)
Comment (vhavalagi) [reply ibm/service/appconfiguration/resource_ibm_app_config_feature.go]: Addressed the above comments

**Pull request merged**